transport/src/main/java/io/netty/channel/PendingWritesTrackerFactory.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.channel;
+
+public final class PendingWritesTrackerFactory {
```

Comment:
Package private for now please

Reply:
If I make this with package visibility then the `io.netty.handler.ssl.SslHandler.SslHandlerCoalescingBufferQueue` will not be able to access it.

Reply:
maybe annotate with ```@UnstableApi``` then?

Reply:
Alternative it could be duplicated for the time being, but there should be a generic contract for any conflation/aggregation that correctly sync channel writeability.